python/hsfs/core/vector_server.py

```diff
@@ -15,319 +15,196 @@
+        ] = None,
+        training_dataset_version: Optional[int] = None,
+        serving_keys: Optional[List[hsfs.serving_key.ServingKey]] = None,
+        skip_fg_ids: Optional[Set] = None,
     ):
+        self._training_dataset_version = training_dataset_version
+        self._feature_store_id = feature_store_id
+
         if features is None:
             features = []
-        self._training_dataset_version = training_dataset_version
+
         self._features = features
-        self._feature_vector_col_name = (
-            [
-                feat.name
-                for feat in features
-                if not (
-                    feat.label
-                    or feat.inference_helper_column
-                    or feat.training_helper_column
-                )
-            ]
-            if features
-            else []
-        )
+        self._feature_vector_col_name = [
+            feat.name
+            for feat in features
+            if not (
+                feat.label
+                or feat.inference_helper_column
+                or feat.training_helper_column
+            )
+        ]
+        self._inference_helper_col_name = [
+            feat.name for feat in features if feat.inference_helper_column
+        ]
+
         self._skip_fg_ids = skip_fg_ids or set()
-        self._prepared_statement_engine = None
-        self._prepared_statements = None
-        self._helper_column_prepared_statements = None
         self._serving_keys = serving_keys
-        self._valid_serving_key = None
-        self._pkname_by_serving_index = None
-        self._prefix_by_serving_index = None
-        self._serving_key_by_serving_index = {}
-        self._external = True
-        self._feature_store_id = feature_store_id
-        self._training_dataset_api = training_dataset_api.TrainingDatasetApi(
-            feature_store_id
-        )
-        self._feature_view_api = feature_view_api.FeatureViewApi(feature_store_id)
-        self._storage_connector_api = storage_connector_api.StorageConnectorApi()
+
         self._transformation_function_engine = (
             transformation_function_engine.TransformationFunctionEngine(
                 feature_store_id
             )
         )
-        self._feature_view_engine = feature_view_engine.FeatureViewEngine(
-            feature_store_id
-        )
         self._transformation_functions = None
         self._required_serving_keys = None
-        self._async_pool = None
+        self._online_store_sql_client = None
 
     def init_serving(
         self,
-        entity,
-        batch,
-        external,
-        inference_helper_columns=False,
-        options=None,
+        entity: Union[feature_view.FeatureView, training_dataset.TrainingDataset],
+        external: bool = False,
+        inference_helper_columns: bool = False,
+        options: Optional[Dict[str, Any]] = None,
     ):
         if external is None:
```

Comment:
since we made external as bool in argument, this check does not work as the if is never satisfied. So if the user did not specify `external` argument in `init_serving` it will default to False, even if using an external client. I am not sure why we had this check but currently we can remove this code as we expect the user to specify the external flag always. Or we just remove the argument from the signature and always derive the external flag using the client.

Reply:
No we should not do that, some people run weird stuff on their cluster. I will fix the check

Reply:
Nice job seeing it